packages/components/src/components/clipboard/clipboard.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import Popover from '../popover';
+import Icon from '../icon';
+
+const Clipboard = ({ text_copy, info_message, success_message, className, popoverClassName }) => {
+    const [is_copied, setIsCopied] = React.useState(false);
+
+    const copyToClipboard = text => {
+        const textField = document.createElement('textarea');
+        textField.innerText = text;
+        document.body.appendChild(textField);
+        textField.select();
+        document.execCommand('copy');
+        textField.remove();
+    };
+
+    const onClick = () => {
+        copyToClipboard(text_copy);
+        setIsCopied(true);
+        setTimeout(() => {
```

Comment:
is it possible to get the value of `setTimeout` here and cancel it with `useEffect`? It is rare but in the case unmounting this component sooner than setTimeout, we are going to hit `setting state on unmounted component error`, right? :thinking:

Reply:
thats right, yes, should be possible, let me try 🙏